plugins/modules/win_partition.ps1

```diff
@@ -100,7 +100,7 @@ if ($null -ne $disk_number -and $null -ne $partition_number) {
 # Check if drive_letter is either auto-assigned or a character from A-Z
 elseif ($drive_letter -and $drive_letter -ne "auto" -and -not ($disk_number -and $partition_number)) {
     if ($drive_letter -match "^[a-zA-Z]$") {
-        $ansible_partition = Get-Partition -DriveLetter $drive_letter -ErrorAction SilentlyContinue
+        $ansible_partition = Get-Partition -DriveLetter $drive_letter -ErrorAction SilentlyContinue | Where-Object { $null -ne $_.DiskNumber }
```

Comment:
Same thing here, can you explain how this fixes the issue? When would there be a partition without a DiskNumber and why should we be ignoring them?